orttraining/orttraining/training_ops/cpu/tensor/gather_grad.cc

```diff
@@ -71,21 +73,24 @@ Status GatherGrad::ComputeImpl(const TensorShape& data_shape, const Tensor& indi
     }
   }
 
-#ifdef USE_OPENMP
-#pragma omp parallel for
-#endif
-  for (int64_t g = 0; g < grad_size; g++) {
+  std::mutex mtx;
+  auto lambda = [&](int64_t g) {
```

Comment:
nit. Personal preference, loop body seems short enough that it can stay inside the for loop.